tests/std/tests/P1522R1_difference_type/test.cpp

```diff
@@ -35,9 +35,133 @@ namespace ordtest {
+                    if (digit == static_cast<unsigned int>(-1)) {
+                        return {u128_parse_status::invalid, std::_Unsigned128{}};
+                    }
+
+                    if (val > u128_max / Base || Base * val > u128_max - digit) {
+                        return {u128_parse_status::overflow, std::_Unsigned128{}};
+                    }
+
+                    val = Base * val + digit;
+                }
+                return {u128_parse_status::valid, val};
+            }
+        };
+
+        template <unsigned int Base>
+        struct parse_u128_impl<Base> {
+            NODISCARD static CONSTEVAL u128_parse_result parse() noexcept {
+                return {u128_parse_status::valid, 0};
+            }
+        };
+
+        template <char... Chars>
+        struct parse_u128 : parse_u128_impl<10, Chars...> {};
+
+        template <char... Chars>
+        struct parse_u128<'0', 'X', Chars...> : parse_u128_impl<16, Chars...> {};
+
+        template <char... Chars>
+        struct parse_u128<'0', 'x', Chars...> : parse_u128_impl<16, Chars...> {};
+
+        template <char... Chars>
+        struct parse_u128<'0', 'B', Chars...> : parse_u128_impl<2, Chars...> {};
+
+        template <char... Chars>
+        struct parse_u128<'0', 'b', Chars...> : parse_u128_impl<2, Chars...> {};
+
+        template <char... Chars>
+        struct parse_u128<'0', Chars...> : parse_u128_impl<8, Chars...> {};
+    } // namespace i128_udl_detail
+
+    template <char... Chars>
+    NODISCARD CONSTEVAL std::_Unsigned128 operator"" _u128() noexcept {
+        constexpr auto parsed_result = i128_udl_detail::parse_u128<Chars...>::parse();
+        static_assert(parsed_result.status_code != i128_udl_detail::u128_parse_status::invalid,
+            "Invalid characters in the integer literal");
+        static_assert(parsed_result.status_code != i128_udl_detail::u128_parse_status::overflow,
+            "The integer literal is too large for an unsigned 128-bit number");
+        return parsed_result.value;
+    }
+
+    template <char... Chars>
+    NODISCARD CONSTEVAL std::_Signed128 operator"" _i128() noexcept {
+        constexpr auto parsed_result = i128_udl_detail::parse_u128<Chars...>::parse();
+        static_assert(parsed_result.status_code != i128_udl_detail::u128_parse_status::invalid,
+            "Invalid characters in the integer literal");
+        static_assert(parsed_result.status_code != i128_udl_detail::u128_parse_status::overflow
+                          && parsed_result.value._Word[1] < (static_cast<std::uint64_t>(1) << 63),
+            "The integer literal is too large for a signed 128-bit number");
+        return static_cast<std::_Signed128>(parsed_result.value);
+    }
+} // namespace i128_literals
+
 using std::_Signed128;
 using std::_Unsigned128;
-using namespace std::literals::_Int128_literals;
+
+using i128_literals::operator""_i128;
+using i128_literals::operator""_u128;

```

Comment:
Why put these in a namespace and then pull them back out instead of defining them in the global namespace in the first place?